### ISSUE TITLE
[BUG?] Wrong Results with other mask_types using encode.py and decode.py

### ISSUE THREAD

**Issue by chunbaobao** [BYchao100/Deep-Lossy-Plus-Residual-Coding#8]
Hi, it's an excellent piece of work.

Everything worked well when the `mask_type` was set to the default, which is `'3P'`. However, when I set the `mask_type` to other values, the results were as expected for `mask_type` values `'4P'` and `'5P'`. Unfortunately, it failed for `mask_type` values `'P'` and `'2P'`. 

Could this issue be because the training process was conducted under the `'3P'` `mask_type`? Does it metter when the `mask_type` are different between training and runtime? Below are the modified code snippets for [`encode.py`](https://github.com/BYchao100/Deep-Lossy-Plus-Residual-Coding/blob/main/DLPR_ll/encode.py) and [`decode.py`](https://github.com/BYchao100/Deep-Lossy-Plus-Residual-Coding/blob/main/DLPR_ll/decode.py):

```
    ### encode

    mask_type = '2P'
    patch_sz = 64
    ll_module = LosslessCompressor(192, mask_type=mask_type).eval().to(device)

    ckp = torch.load(os.path.join(ckp_dir, "ckp.tar"), map_location=device)
    ll_module.load_state_dict(ckp['model_state_dict'])
    ll_module.lossy_compressor.update(force=True)

    # COT
    COT = coding_order_table7x7(patch_sz=patch_sz, mask_type=mask_type)
```

```
    ### decode
    mask_type = '2P'
    ll_module = LosslessCompressor(192, mask_type=mask_type).eval().to(device)

    ckp = torch.load(os.path.join(ckp_dir, "ckp.tar"), map_location=device)
    ll_module.load_state_dict(ckp['model_state_dict'])
    ll_module.lossy_compressor.update(force=True)

    COT = coding_order_table7x7(mask_type=mask_type)

```
The results for `mask_type` values `'P'` and `'2P'` were as follows:
![temp22](https://github.com/user-attachments/assets/3be60fb3-9954-428f-9611-249e81515232)
![temp2](https://github.com/user-attachments/assets/0487ba81-9399-4925-b927-8687e2a61c05)

**Comment by BYchao100** [comment]
In order to use other mask types, the model must be trained under the specific mask type. We only provide the `3P` model, which is selected as the final model.

**Comment by chunbaobao** [comment]
Thank you for your prompt reply. I am still somewhat unclear about this. From my understanding, a poorly trained model results in inaccurate predictions of the data distribution, which leads to a high compression rate, rather than incorrect encoding/decoding. Changing the mask_type during runtime seems to only affect the context information of a single pixel, which may result in poor compression rather than incorrect encoding/decoding. Could I be missing some important details that would explain why the decoding result is incorrect?

**Comment by chunbaobao** [comment]
By the way, '4P' '5P' and ‘S' mask_type will lead to correct encode/decoding.
